Sources/Target/Linux/Process.cpp

```diff
@@ -232,48 +232,54 @@ ErrorCode Process::wait() {
-      // stopped by a clone but still stepping while encountering a kEventNone.
-      // This should not happen and we just assert that it does not.
-
-      if (_currentThread->_stopInfo.reason == StopInfo::kReasonThreadSpawn &&
-          stepping) { //(2b)
-        unsigned long spawnedThreadIdData;
-        CHK(ptrace().getEventMessage(_pid, spawnedThreadIdData));
-        auto const spawnedThreadId = static_cast<ThreadId>(spawnedThreadIdData);
-
-        if (_threads.find(spawnedThreadId) == _threads.end()) {
-          // If the newly cloned thread does not yet exist within the _threads
-          // vector then we have not yet seen it with waitpid(2). The most
-          // recent
-          // waitpid(2) returns a status guaranteeing a clone event did happen.
-          // Therefore, we wait and block until we see the cloned thread.
-          int spawnedThreadStatus;
-          ThreadId returnedThreadId =
-              blocking_waitpid(spawnedThreadId, &spawnedThreadStatus, __WALL);
-          if (spawnedThreadId != returnedThreadId)
-            return kErrorProcessNotFound;
-          DS2LOG(Debug, "child thread tid %d %s", returnedThreadId,
-                 Stringify::WaitStatus(spawnedThreadStatus));
-
-          _currentThread->step();
-          auto newThread = new Thread(this, returnedThreadId);
-          newThread->beforeResume();
-        } else {
-          // The new thread corresponding to this kReasonThreadSpawn was already
-          // caught by waitpid(2). Therefore, we only need to step the current
-          // thread.
-          _currentThread->step();
+
+      switch (_currentThread->_stopInfo.reason) {
+      case StopInfo::kReasonNone: // (1)
+        if (stepping)
+          CHK(_currentThread->step());
+        else
+          CHK(_currentThread->resume());
+        break;
+
+      case StopInfo::kReasonThreadSpawn:
+        if (!stepping) // (2a)
+          CHK(_currentThread->resume());
+        else { // (2b)
+          unsigned long spawnedThreadIdData;
+          CHK(ptrace().getEventMessage(_pid, spawnedThreadIdData));
+          auto const spawnedThreadId = static_cast<ThreadId>(spawnedThreadIdData);
+          if (_threads.find(spawnedThreadId) == _threads.end()) {
+            // If the newly cloned thread does not yet exist within the _threads
+            // vector then we have not yet seen it with waitpid(2). The most
+            // recent
+            // waitpid(2) returns a status guaranteeing a clone event did happen.
+            // Therefore, we wait and block until we see the cloned thread.
+            int spawnedThreadStatus;
+            ThreadId returnedThreadId =
+                blocking_waitpid(spawnedThreadId, &spawnedThreadStatus, __WALL);
+            if (spawnedThreadId != returnedThreadId)
+              return kErrorProcessNotFound;
+            DS2LOG(Debug, "child thread tid %d %s", returnedThreadId,
+                  Stringify::WaitStatus(spawnedThreadStatus));
```

Comment:
Can't tell if GH or the indentation is off.

Reply:
Indentation is off by 1 space; fixing.